flink-runtime/src/test/java/org/apache/flink/runtime/state/ChannelPersistenceITCase.java

```diff
@@ -0,0 +1,179 @@
+				writer.addInputData(checkpointId, e.getKey(), SEQUENCE_NUMBER_UNKNOWN, e.getValue());
+			}
+			writer.finishInput(checkpointId);
+			for (Map.Entry<ResultSubpartitionInfo, Buffer> e : rsBuffers.entrySet()) {
+				writer.addOutputData(checkpointId, e.getKey(), SEQUENCE_NUMBER_UNKNOWN, e.getValue());
+			}
+			writer.finishOutput(checkpointId);
+			ChannelStateWriteResult result = writer.getWriteResult(checkpointId);
+			result.getResultSubpartitionStateHandles().join(); // prevent abnormal complete in close
+			return result;
+		}
+	}
+
+	public static CheckpointStorageWorkerView getStreamFactoryFactory() {
+		return getStreamFactoryFactory(42);
+	}
+
+	public static CheckpointStorageWorkerView getStreamFactoryFactory(int maxStateSize) {
+		return new CheckpointStorageWorkerView() {
+			@Override
+			public CheckpointStreamFactory resolveCheckpointStorageLocation(long checkpointId, CheckpointStorageLocationReference reference) {
+				return new NonPersistentMetadataCheckpointStorageLocation(maxStateSize);
+			}
+
+			@Override
+			public CheckpointStreamFactory.CheckpointStateOutputStream createTaskOwnedStateStream() {
+				throw new UnsupportedOperationException();
+			}
+		};
+	}
+
+	private byte[] read(TaskStateSnapshot taskStateSnapshot, int size, BiFunctionWithException<ChannelStateReader, MemorySegment, ReadResult, Exception> readFn) throws Exception {
+		byte[] dst = new byte[size];
+		HeapMemorySegment mem = HeapMemorySegment.FACTORY.wrap(dst);
+		try {
+			checkState(NO_MORE_DATA == readFn.apply(new ChannelStateReaderImpl(taskStateSnapshot), mem));
+		} finally {
+			mem.free();
+		}
+		return dst;
+	}
+
+	private TaskStateSnapshot toTaskStateSnapshot(ChannelStateWriteResult t) throws Exception {
+		return new TaskStateSnapshot(singletonMap(new OperatorID(),
+			new OperatorSubtaskState(
+				StateObjectCollection.empty(),
+				StateObjectCollection.empty(),
+				StateObjectCollection.empty(),
+				StateObjectCollection.empty(),
+				new StateObjectCollection<>(t.getInputChannelStateHandles().get()),
+				new StateObjectCollection<>(t.getResultSubpartitionStateHandles().get())
+			)
+		));
+	}
+
+	private <C> List<C> collect(Collection<StateObject> handles, Class<C> clazz) {
+		//noinspection unchecked
+		return handles.stream().filter(clazz::isInstance).map(h -> (C) h).collect(Collectors.toList());
+	}
+
+	private static int sizeOfBytes(Map<?, byte[]> map) {
+		return map.values().stream().mapToInt(d -> d.length).sum();
+	}
+
+	private <K> Map<K, Buffer> wrapWithBuffers(Map<K, byte[]> icMap) {
+		return icMap.entrySet().stream().collect(Collectors.toMap(Map.Entry::getKey, e -> wrapWithBuffer(e.getValue())));
+	}
+
+	private static Buffer wrapWithBuffer(byte[] data) {
+		NetworkBuffer buffer = new NetworkBuffer(HeapMemorySegment.FACTORY.allocateUnpooledSegment(data.length, null), FreeingBufferRecycler.INSTANCE);
+		buffer.writeBytes(data);
+		return buffer;
+	}
+
```

Comment:
you should decide on empty line or not?

Reply:
I don't think it makes a big difference. If it would, it should be in the Flink formatting guidelines and checkstyle config.